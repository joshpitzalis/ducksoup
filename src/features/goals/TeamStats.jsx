import React from 'react';
import { Sparklines, SparklinesLine } from 'react-sparklines';

export const TeamStats = () => {
  const emailSubscribers = [0, 2, 11];
  const githubIssues = [
    3,
    12,
    13,
    8,
    9,
    10,
    12,
    21,
    19,
    22,
    32,
    46,
    55,
    51,
    71,
    73,
    68,
  ];
  return (
    <section className="pa3 pa5-ns" data-name="slab-stat-small">
      {/* <h3 className="f6 ttu tracked">Project Stats</h3> */}
      <div className="cf">
        <dl className="db dib-l w-auto-l lh-title mr6-l">
          {/* useful for finding github issues count history */}
          {/* https://9-volt.github.io/bug-life/?repo=joshpitzalis/tinyteams */}
          <Sparklines data={githubIssues}>
            <SparklinesLine style={{ fill: 'none' }} />
          </Sparklines>
          <dd className="f6 fw4 ml0">Github Issues</dd>
          <dd className="f7 fw4 ml0 o-30">Last updated 27 March</dd>
          <dd className="f2 f-subheadline-l fw6 ml0">68</dd>
        </dl>
        <dl className="db dib-l w-auto-l lh-title mr6-l">
          <Sparklines data={emailSubscribers}>
            <SparklinesLine style={{ fill: 'none' }} />
          </Sparklines>
          <dd className="f6 fw4 ml0">Email Subscribers</dd>
          <dd className="f7 fw4 ml0 o-30">Last updated 27 March</dd>
          <dd className="f2 f-subheadline-l fw6 ml0">11</dd>
        </dl>
      </div>
    </section>
  );
};
