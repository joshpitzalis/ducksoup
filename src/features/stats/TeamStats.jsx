import React from 'react';

export const TeamStats = () => (
  <section className="pa3 pa5-ns" data-name="slab-stat-small">
    <h3 className="f6 ttu tracked">Today</h3>
    <div className="cf">
      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title mr5-l">
        <dd className="f6 fw4 ml0">Votes cast</dd>
        <dd className="f3 fw6 ml0">24</dd>
      </dl>
      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title mr5-l">
        <dd className="f6 fw4 ml0">Todos created</dd>
        <dd className="f3 fw6 ml0">3</dd>
      </dl>
      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title mr5-l">
        <dd className="f6 fw4 ml0">Hour Worked</dd>
        <dd className="f3 fw6 ml0">44</dd>
      </dl>
      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title mr5-l">
        <dd className="f6 fw4 ml0">Chat Comments</dd>
        <dd className="f3 fw6 ml0">102</dd>
      </dl>

      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title mr5-l">
        <dd className="f6 fw4 ml0">People</dd>
        <dd className="f3 fw6 ml0">3</dd>
      </dl>
      <dl className="fl fn-l w-50 dib-l w-auto-l lh-title">
        <dd className="f6 fw4 ml0">Days left</dd>
        <dd className="f3 fw6 ml0">126</dd>
      </dl>
    </div>
  </section>
);
