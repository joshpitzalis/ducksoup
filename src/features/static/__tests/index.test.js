import 'jest-dom/extend-expect';
import React from 'react';
import { cleanup, fireEvent, render } from 'react-testing-library';
import { Static } from '../index';
import { TeamStats } from '../TeamStats';

afterEach(cleanup);

it('Team stats snapshot', () => {
  const { asFragment } = render(<TeamStats />);
  expect(asFragment()).toMatchSnapshot();
});

it.skip('my little head shows up in the static section when I sign in', async () => {
  const { queryByTestId, getByTestId } = render(<Static />);
  getByTestId('goalRoad');
  expect(queryByTestId('detailsBox')).toBeNull();
  fireEvent.click(getByTestId('objective'));
  getByTestId('detailsBox');
});

// make my head show up in users
// make stats show up on static

// make timeline appear in tasks
