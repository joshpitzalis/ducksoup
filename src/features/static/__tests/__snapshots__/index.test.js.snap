// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team stats snapshot 1`] = `
<DocumentFragment>
  <section
    class="pa5 w-100 pb0"
    data-name="slab-stat-small"
  >
    <div
      class=""
    >
      <dl
        class="db dib-l w-auto-l lh-title mr6-l"
      >
        <svg
          preserveAspectRatio="none"
          viewBox="0 0 240 60"
        >
          <g>
            <circle
              cx="2"
              cy="58"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="16.75"
              cy="50.800000000000004"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="31.5"
              cy="50"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="46.25"
              cy="54"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="61"
              cy="53.2"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="75.75"
              cy="52.400000000000006"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="90.5"
              cy="50.800000000000004"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="105.25"
              cy="43.6"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="120"
              cy="45.2"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="134.75"
              cy="42.800000000000004"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="149.5"
              cy="34.800000000000004"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="164.25"
              cy="23.6"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="179"
              cy="16.4"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="193.75"
              cy="19.6"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="208.5"
              cy="3.6"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="223.25"
              cy="2"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="238"
              cy="6"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <polyline
              points="2 58 16.75 50.800000000000004 31.5 50 46.25 54 61 53.2 75.75 52.400000000000006 90.5 50.800000000000004 105.25 43.6 120 45.2 134.75 42.800000000000004 149.5 34.800000000000004 164.25 23.6 179 16.4 193.75 19.6 208.5 3.6 223.25 2 238 6 238 58 2 58 2 58"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <polyline
              points="2 58 16.75 50.800000000000004 31.5 50 46.25 54 61 53.2 75.75 52.400000000000006 90.5 50.800000000000004 105.25 43.6 120 45.2 134.75 42.800000000000004 149.5 34.800000000000004 164.25 23.6 179 16.4 193.75 19.6 208.5 3.6 223.25 2 238 6"
              style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
            />
          </g>
        </svg>
        <dd
          class="f6 fw4 ml0"
        >
          Github Issues
        </dd>
        <dd
          class="f7 fw4 ml0 o-30"
        >
          Last updated 27 March
        </dd>
        <dd
          class="f2 f-subheadline-l fw6 ml0"
        >
          68
        </dd>
      </dl>
      <dl
        class="db dib-l w-auto-l lh-title mr6-l"
      >
        <svg
          preserveAspectRatio="none"
          viewBox="0 0 240 60"
        >
          <g>
            <circle
              cx="2"
              cy="58"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="80.66666666666667"
              cy="50.53333333333333"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="159.33333333333334"
              cy="16.933333333333334"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <circle
              cx="238"
              cy="2"
              r="2"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <polyline
              points="2 58 80.66666666666667 50.53333333333333 159.33333333333334 16.933333333333334 238 2 238 58 2 58 2 58"
              style="stroke: none; stroke-width: 0; fill-opacity: .1; fill: none; pointer-events: auto;"
            />
            <polyline
              points="2 58 80.66666666666667 50.53333333333333 159.33333333333334 16.933333333333334 238 2"
              style="stroke: slategray; stroke-width: 1; stroke-linejoin: round; stroke-linecap: round; fill: none;"
            />
          </g>
        </svg>
        <dd
          class="f6 fw4 ml0"
        >
          Email Subscribers
        </dd>
        <dd
          class="f7 fw4 ml0 o-30"
        >
          Last updated 27 March
        </dd>
        <dd
          class="f2 f-subheadline-l fw6 ml0"
        >
          11
        </dd>
      </dl>
    </div>
  </section>
</DocumentFragment>
`;
